In [1]:
import pandas as pd
import matplotlib
import random

In [2]:
df = pd.read_csv("../data/test/full_test.csv")
df

,id,title,selftext,multilabel_clf_label,disorder,text,multiclass_clf_label,depression_label,anxiety_label
0,8es6qn,I cut myself for the first time in a year toda...,... and hated that I still loved it. The burni...,"[1, 0]",{'depressive_disorder'},I cut myself for the first time in a year toda...,Depression,1,0
1,ar89pm,Anybody hate spring/summer?,".....And even so, when depression is not too s...","[1, 1]","{'depressive_disorder', 'anxiety_disorder'}","Anybody hate springsummer. And even so, when d...",Comorbid (Depression + Anxiety),1,1
2,5g4v1n,So I'm staring at this tablet of Lexapro...,...and I'm not sure what to do. I guess I don'...,"[1, 1]","{'depressive_disorder', 'anxiety_disorder'}",So Im staring at this tablet of Lexapro. and I...,Comorbid (Depression + Anxiety),1,1
3,9x8h3g,"Found a poem sort of thing, very emo (ha ha ha...","""I am delicate and bitter. I am sweet on the o...","[1, 0]",{'depressive_disorder'},"Found a poem sort of thing, very emo ha ha ha ...",Depression,1,0
4,es440i,You don't get it,"""It gets better"" ""Stop thinking about it"" ""get...","[1, 1]","{'depressive_disorder', 'anxiety_disorder'}",You dont get it. It gets better Stop thinking ...,Comorbid (Depression + Anxiety),1,1
...,...,...,...,...,...,...,...,...,...
2867,lm7uix,My life just feels like this one big party Im ...,You know the feeling. Where youre just so over...,"[1, 0]",{'depressive_disorder'},My life just feels like this one big party Im ...,Depression,1,0
2868,ac1aj9,Just move away from your parents!,You probably want to hear my sob story as much...,"[0, 0]",{'control_group'},Just move away from your parents. You probably...,Normal,0,0
2869,hng1f5,That feeling when...,"You wake up, and that itself just brings you s...","[0, 0]",{'control_group'},"That feeling when. You wake up, and that itsel...",Normal,0,0
2870,4pfk9x,Life just keeps getting more complicated,"You would think after years of bad breaks, som...","[0, 0]",{'control_group'},Life just keeps getting more complicated. You ...,Normal,0,0


In [3]:
control_df = df[(df['depression_label'] == 0) & (df['anxiety_label'] == 0)].reset_index(drop=True)
depression_df = df[(df['depression_label'] == 1) & (df['anxiety_label'] == 0)].reset_index(drop=True)
anxiety_df = df[(df['depression_label'] == 0) & (df['anxiety_label'] == 1)].reset_index(drop=True)
comorbidity_df = df[(df['depression_label'] == 1) & (df['anxiety_label'] == 1)].reset_index(drop=True)
print(f"control_df: {len(control_df)}")
print(f"depression_df: {len(depression_df)}")
print(f"anxiety_df: {len(anxiety_df)}")
print(f"comorbidity_df: {len(comorbidity_df)}")

control_df: 1201
depression_df: 970
anxiety_df: 92
comorbidity_df: 609


In [23]:
control = control_df.sample()
print(control.index[0])
control['text'].tolist()[0], (control['depression_label'].tolist()[0], control['anxiety_label'].tolist()[0])

244


('I hate it when people say you have so much to look forward to. I hate it when I hear, because I know its not true. I have nothing going for me to get excited about or hopeful for. Im pigeonholding a job that doesnt pay enough, Ive flunked out of school, now realizing I wasted my time because I didnt even know what I was in school for. Ive loss passion for making art, and belief in myself. I hate it when I hear that statement from people who dont understand that I really dont have anything to look forward to.',
 (0, 0))

In [34]:
depression = depression_df.sample()
print(depression.index[0])
depression['text'].tolist()[0], (depression['depression_label'].tolist()[0], depression['anxiety_label'].tolist()[0])

393


('extreme pain and depression. I have debilitating menstrual cramps each couple months where I lay in bed for a couple hours, coating in sweat, bracing myself through waves of excruciating pain and at the same time wanting to throw up. I also have depression. The pain is so awful that I want to die. Theres times where Ive laid there, too exhausted to cry, and I repeat over and over again in my head I want to die. How to deal with extreme pain whilst not wanting to kill yourself?',
 (1, 0))

In [61]:
anxiety = anxiety_df.sample()
print(anxiety.index[0])
anxiety['text'].tolist()[0], (anxiety['depression_label'].tolist()[0], anxiety['anxiety_label'].tolist()[0])

8


('What just happened to me. Around 940 pm I felt a sudden wave of intense anxiety wash over me. I had to lock all the doors and when I looked outside i swear I saw someone Theyd have to be over 7 feet tall, but I swear I saw someone. I ran to tell my parents, my dad went out and saw no one. Normally, this would ease people right? It didnt ease me. I felt and still feel, as if I am in danger or going to die. I couldnt stop shaking and I just now 10 pm was able to stop crying. My chest and head hurt really bad now. Was this a really bad anxiety attack or what?',
 (0, 1))

In [126]:
comorbidity = comorbidity_df.sample()
print(comorbidity.index[0])
comorbidity['text'].tolist()[0], (comorbidity['depression_label'].tolist()[0], comorbidity['anxiety_label'].tolist()[0])

203


('Found out this place. Hey. So, I was always posting in rdepression, and I saw there was this subreddit and I thought I might give it a try. My life is hell now, completely pointless, completely. hell. and I dunno if in one week I will still be alive. I try to give life a chance. I try to live life day by day, and now, it only seems like its been downhill for a while, and nothing is going well and nothing ever will. I mean, Im one in billions of humans, my life is meaningless, pointless and I dont matter. Why keep on going when you get your heart broken, hurt people, lose your job, get in shit with your whole family, cant even find happiness in small things? Cant sleep well, cant breathe, the earth is burning and no ones cares about my pain, me a meaningless human? WHY WAS I GIVEN THIS LIFE IF HAPPINESS IS SHORT AND FULL OF BULLSHIT AND YOU HAVE TO WORK HARD TO OBTAIN IT? WHY CANT I JUST DIE? WHY DO I KEEP GIVING LIFE CHANCES BUT IT KEEPS DISAPPOINTING ME EVERYDAY? IT HURTS, I WANT TO

# Control Example:

Does anyone else have this trigger or a similar trigger. so this morning when i woke up the first thing i saw was this huge ass bug on the floor and i just stood there, literally frozen. had to get my mom to kill it lol. anyway, all day ive been feeling more bugs crawl on me than usual. i even felt a bug literally under my eyelid it was horrible. i also rarely feel bugs under my skin but i felt that a bit today. also ive noticed if i think about the bugs, more appear. i only slept maybe 2 hours and pull all nighters on accident a lot, can sleep deprivation cause more hallucinations than normal?


I dont hate myself as much as before. I used to hate myself for being ugly , having shitty personality and a few mental problems , but since I accepted it I feel a bit better . I know I will never have good friend ,good job , be loved or find girlfriend , but I dont feel bad anymore , because I know how boring I am and my looks are pretty bad too. The only think that bothers me is that I am more stupid than I would like to admit


# Depression Example

Tempted to self harm. I was diagnosed with depression about 7 months ago. At times it feels like it has gotten better but then it feels like its gotten worse. For the past two weeks I have felt a strong urge to self harm. I dont know why since I have never done it in the first place. I dont understand where this urge is coming from but it literally keeps me up at night. I get tempted to do it but eventually I become exhausted and fall asleep. Im a wuss for any sort of physical pain which has me even more confused as to why I want to hurt myself. Can anyone relate to what Im saying. Am I crazy?

Im tired of being like this. I know it sounds so melodramatic, but everyday, I get more and more tired of being so unhappy. I was diagnosed with depression and I take the meds prescribed to me. I also do other things to busy myself, and reconnect with people that I had stopped talking with during my dark days. Yet, I struggle to perform tasks with real vigor and Im being secretive of my feelings again. I even start to fake sickness again just to avoid the world for a day. Also, I fake happiness, and when Im alone with my thoughts, I realize that I am still not improving. I dont like being depressed and I hate myself so much. I dont want to be like this. I cant be. What do I do? How can I stop myself?


I live in constant pain. I mean seriously, whats the point of being alive? I literally suffer constant emotional anguish every single day and the best i can do is distract myself from it. Most of the time i just want to curl up into a ball and cry my eyes out but im so disconnected i cant cry anymore. I mean im not going to kill myself but how can anyone seriously expect me to want to keep living when every day is torture? Nothing ever helps nothing ever makes it better and i can feel myself slowly getting worse each day. I wish i could disappear

# Anxiety Example

Does anybody else have trouble expressing anger. All my life I have had trouble externalizing anger out of fear that people are going to leave me so now everytime I get angry, which is often, legitimate reason or not, I tell myself I am crazy so I never express to other people what angered me and its eating me alive. At most, I start getting somatizations and shaking but thats about it. Does anyone else feel this way? Does anyone else have tips on how to tell people things that bother youmake you angry without fearing you will explode?

What just happened to me. Around 940 pm I felt a sudden wave of intense anxiety wash over me. I had to lock all the doors and when I looked outside i swear I saw someone Theyd have to be over 7 feet tall, but I swear I saw someone. I ran to tell my parents, my dad went out and saw no one. Normally, this would ease people right? It didnt ease me. I felt and still feel, as if I am in danger or going to die. I couldnt stop shaking and I just now 10 pm was able to stop crying. My chest and head hurt really bad now. Was this a really bad anxiety attack or what?


# Comorbidity Example

Stuck after college. I graduated back in August and Ive yet to get a job in my field. I still work in my shitty retail job a few times a week and Im so severely depressed and lonely I am frozen with fear of applying to as many jobs as I should due to feelings of inadequacy and fear. Its killing me. I spend everyday in a constant cycle of loneliness and anxiety and worry and panic and emptiness. Everyday I feel like more and more of a burden to my family for not finding a job and starting my life. I just want this years long suffering to end and its only getting worse. I hate myself for it, Ive withdrawn from my friendships and Im dreading Christmas because of family questions about why Im not working yet.